src/wfs/src/main/java/org/geoserver/wfs/GetFeature.java

```diff
@@ -1229,7 +1229,21 @@ void encodeQueryAsKvp(
         Filter transformedFilter = filter;
         if (declaredCRS != null) {
             transformedFilter =
-                    WFSReprojectionUtil.normalizeFilterCRS(filter, source.getSchema(), declaredCRS);
+                    WFSReprojectionUtil.normalizeFilterCRS(
+                            filter, source.getSchema(), declaredCRS, declaredCRS);
+        } else {
+            // let's try to sue the feature type info CRS
+            FeatureTypeInfo featureTypeInfo =
+                    catalog.getFeatureTypeByName(
+                            primaryTypeName.getPrefix(), primaryTypeName.getLocalPart());
+            if (featureTypeInfo != null && featureTypeInfo.getCRS() != null) {
+                // the feature type info has a CRS defined, so let's use it
+                declaredCRS =
```

Comment:
So here you're looking at handling axis flipping, but it does not happen in the path above?

Reply:
Makes sense